### ISSUE TITLE
[Uno] Using mapsui with .NET 8 causes invalid targets resolution on Windows

### ISSUE THREAD

**Issue by jeromelaban** [Mapsui/Mapsui#2523]
**Mapsui Version**
4.1.4

**Mapsui Platform**
All of them, using `Mapsui.Uno.WinUI`

**Describe the bug**
Using a default uno platform template, including the WinAppSDK head, the build will choose to use `net8.0` instead of using `net7.0-windows10.0.19041`, which is incorrect on WinAppSDK. This is caused by a behavior of NuGet which favors the closest major version a non-platform specific TFM even when there's a matching non-platform specific version with a previous major version.

**To Reproduce**
1. Create a blank template
2. Add mapsui to the shared project
3. Add the maps ui control to `mainpage.xaml`
4. Build and the following error will appear 
   ```
   Error WMC1006 Cannot resolve Assembly or Windows Metadata file 'Type universe cannot resolve assembly: Uno.UI,
   ```

**Workaround**
The following inclusion can be used instead:

```xml
  <ItemGroup Condition="'$(TargetFramework)'=='$(DotNetVersion)-windows10.0.19041'">
    <PackageReference Include="Mapsui.Uno.WinUI" ExcludeAssets="Compile" GeneratePathProperty="true" />
    <PackageReference Include="Mapsui" GeneratePathProperty="true" />

    <Reference Include="Mapsui.UI.Uno.WinUI">
      <HintPath>$(PkgMapsui_Uno_WinUI)\lib\net7.0-windows10.0.19041\Mapsui.UI.Uno.WinUI.dll</HintPath>
    </Reference>
  </ItemGroup>
  <ItemGroup Condition="'$(TargetFramework)'!='$(DotNetVersion)-windows10.0.19041'">
    <PackageReference Include="Mapsui.Uno.WinUI" />
  </ItemGroup>
```

Which will avoid including the wrong dll version in the uno common project.

**Additional context**
The fix for this issue is to add a `net8.0-windows10.0.19041` target in the nuget package, in order for nuget to pick the right version.

Originally posted on https://github.com/unoplatform/uno/discussions/15652 by @CaymanIslands.

**Comment by jeromelaban** [comment]
@inforithmics thanks for looking into this!

**Comment by pauldendulk** [comment]
This should be fixed with the PR of Thomas that is now released in 4.0.5.
